src/ILCompiler.WebAssembly/src/CodeGen/EvaluationStack.cs

```diff
@@ -181,18 +196,58 @@ internal abstract class StackEntry
         /// </summary>
         public TypeDesc Type { get; }
 
-        public LLVMValueRef LLVMValue { get; set; }
+        Dictionary<LLVMTypeRef, LLVMValueRef> _castValues = new Dictionary<LLVMTypeRef, LLVMValueRef>(LLVMTypeRefEquality.Comparer);
+
+        public LLVMValueRef ValueAsType(LLVMTypeRef type, ILImporter importer)
```

Comment:
Adding an ILImporter dependency to StackEntry introduces an unnecessary circular dependency. If you just need the importer for CastAsNecessary, maybe you can just pass a Builder or something